### PR TITLE
Rebuild segmenter testdata and move tests back into segmenter crate

### PR DESCRIPTION
**Pull request by sffc**
I can go either way on whether to make special segmenter testdata (with all locales) or rely on datagen tests, but there were other tests that were disabled because they didn't have access to this data. So I think I lean toward special-casing segmenter and putting its own JSON data in the tests directory.

After discussing with @Manishearth, I think this is a reasonable approach, and we should work toward a future where we can put these special cases directly in testdata or repodata.

### CODE REVIEW

**Review by Manishearth — APPROVED** [review]

**Review by zbraniecki — APPROVED** [review]

**Review by Manishearth — APPROVED** [review]

**Comment by sffc** [comment]
@robertbastian Feel free to merge this once you approve.

**Review by sffc — COMMENTED** [review]
LGTM

Was considering postcard for this reason but went back and forth on it

**Review by robertbastian — APPROVED** [review]

**Pull request merged**
